src/wampy.js

```diff
@@ -927,7 +927,7 @@ class Wampy {
             this._renewSubscriptions();
             this._renewRegistrations();
 
-        } else {
+        } else if (this._cache.connectPromise?.onSuccess) {
```

Comment:
Interesting... How did you get into this situation? `this._cache.connectPromise` is a promise that is always instantiated during `.connect()` so it should exist. If this is an auto reconnection flow — then the above branch should run...

Reply:
It happens if `onError` doesn't interrupt code flow and I also introduce a NoSerializerAvailableError() for example. Since onError will not throw an error, when code reaches line 801: `this._cache.connectPromise = null;` and then when running `_onWelcomeMessage` it will try to access `onSuccess` inside a null object.

```
const wampy = new Wampy(url, { 
   realm: 'realm',
   ws: WebSocket,
   onError: () => {},
});
wsSetProtocol('cbor');

await wampy.connect();
```
it throws:
```
TypeError: Cannot read properties of null (reading 'onSuccess')
```

Reply:
Hm.... So the actual problem is not here.... If Server has chosen serializer, that is not available, then wampy must error and abort ws connection! As I see: now only error callbacks/rejects are processed.
The problematic point is `_wsOnOpen()`. Because if no serializer available — no send/receive messages can be done and of course no welcome!
I think the real cause can be fixed by adding `hardclose()` somewhere around L807 in `_wsOnOpen()`

Reply:
Makes sense! I will convert this to a draft for now and push new changes 👍